# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libgpod/hotfix-gcc14.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Fix build with GCC 14+: explicit casts and typed K&R parameters.

diff -urp a/src/ithumb-writer.c b/src/ithumb-writer.c
--- a/src/ithumb-writer.c	2011-07-24 13:08:03.000000000 +0200
+++ b/src/ithumb-writer.c	2026-09-26 17:06:00.363483658 +0200
@@ -728,7 +728,7 @@ ithumb_writer_handle_rotation (GdkPixbuf
   {
       return gdk_pixbuf_rotate_simple (pixbuf, *rotation);
   }
-  return g_object_ref (G_OBJECT (pixbuf));
+  return (GdkPixbuf*) g_object_ref (G_OBJECT (pixbuf));
 }
 
 /* On the iPhone, thumbnails are presented as squares in a grid.
@@ -968,7 +968,7 @@ ithumb_writer_write_thumbnail (iThumbWri
     else if (thumb->data_type == ITDB_THUMB_TYPE_PIXBUF)
     {
         Itdb_Thumb_Pixbuf *thumb_pixbuf = (Itdb_Thumb_Pixbuf *)thumb;
-        pixbuf = g_object_ref (G_OBJECT (thumb_pixbuf->pixbuf));
+        pixbuf = (GdkPixbuf*) g_object_ref (G_OBJECT (thumb_pixbuf->pixbuf));
     }
 
     if (pixbuf == NULL)
diff -urp a/tools/generic-callout.c b/tools/generic-callout.c
--- a/tools/generic-callout.c	2013-07-10 15:43:42.000000000 +0200
+++ b/tools/generic-callout.c	2026-09-26 17:06:00.363939130 +0200
@@ -664,7 +664,7 @@ static gboolean write_sysinfo_extended (
 }
 
 
-static char *get_info_from_usb (usb_bus_number, usb_device_number)
+static char *get_info_from_usb (guint usb_bus_number, guint usb_device_number)
 {
 #ifdef HAVE_LIBUSB
         return read_sysinfo_extended_from_usb (usb_bus_number,
Only in b/tools: generic-callout.c.orig
